Tabloid/Repositories/CommentRepository.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Tabloid.Utils;
+using Tabloid.Models;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Data.SqlClient;
+
+namespace Tabloid.Repositories
+{
+    public class CommentRepository : BaseRepository, ICommentRepository
+    {
+        public CommentRepository(IConfiguration configuration) : base(configuration) { }
+
+        public List<Comment> GetCommentsByPostId(int id)
+        {
+            using (var conn = Connection)
+            {
+                conn.Open();
+                using (var cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"
+                SELECT c.Id, c.UserProfileId, Subject, c.Content, c.CreateDateTime, u.FirstName, u.LastName, u.DisplayName, 
+                    u.Email, 
+                    u.CreateDateTime AS UserProfileDateCreated,
+                    u.ImageLocation AS AvatarImage,
+                    u.UserTypeId 
+                  FROM Comment c
+                  JOIN Post p on c.PostId = p.Id
+                  Join UserProfile u on p.UserProfileId = u.Id
+              WHERE PostId = @Id";
+
+                    DbUtils.AddParameter(cmd, "@Id", id);
+
+                    var reader = cmd.ExecuteReader();
+
+                    List<Comment> comments = new List<Comment>();
+
+                    while (reader.Read())
+                    {
+                        comments.Add(new Comment()
+                        {
+                            Id = DbUtils.GetInt(reader, "Id"),
+                            PostId = id,
+                            UserProfileId = DbUtils.GetInt(reader, "UserProfileId"),
+                            Subject = DbUtils.GetString(reader, "Subject"),
+                            Content = DbUtils.GetString(reader, "Content"),
+                            CreateDateTime = DbUtils.GetDateTime(reader, "CreateDateTime"),
+                            UserProfile = new UserProfile()
+                            {
+                                Id = DbUtils.GetInt(reader, "UserProfileId"),
+                                FirstName = DbUtils.GetString(reader, "FirstName"),
+                                LastName = DbUtils.GetString(reader, "LastName"),
+                                DisplayName = DbUtils.GetString(reader, "DisplayName"),
+                                Email = DbUtils.GetString(reader, "Email"),
+                                CreateDateTime = DbUtils.GetDateTime(reader, "UserProfileDateCreated"),
+                                ImageLocation = DbUtils.GetString(reader, "AvatarImage"),
+                                UserTypeId = DbUtils.GetInt(reader, "UserTypeId")
+                            }
```

Comment:
You could also make a post object here, you're already joining the post from the comment. You could add the title of the post in the select statement. You might have to also select other parts of the post as well but in this case we only really need the title.

Reply:
Went with js suggestion below in lieu of